infra/gcp/ensure-staging-storage.sh

```diff
@@ -46,6 +46,7 @@ STAGING_PROJECTS=(
     cluster-api
     cluster-api-aws
     cluster-api-azure
+    cluster-api-do
```

Comment:
@detiber Why are some "cluster-api" and others "capi" ?  Is there a principle as to which to use?

Reply:
> I think it's related to the max length of the names, but I'm not sure if it's related to GCR names or GCS Bucket ones?

Yes, I believe the same. `cluster-api-do` is less than 18 characters, so I went with that name.

Reply:
@bartsmykla is correct, the main reason for the bifurcation was due to character length. We'll work to unify the naming as we work on the proposal for nesting.